dataactcore/models/stagingModels.py

```diff
@@ -1,15 +1,261 @@
+    tas4 = tas4 if tas4 else '0000'
+    tas5 = context.current_parameters['availabilitytypecode']
+    tas5 = tas5 if tas5 else ' '
+    tas6 = context.current_parameters['mainaccountcode']
+    tas6 = tas6 if tas6 else '0000'
+    tas7 = context.current_parameters['subaccountcode']
+    tas7 = tas7 if tas7 else '000'
+    tas = '{}{}{}{}{}{}{}'.format(tas1, tas2, tas3, tas4, tas5, tas6, tas7)
+    return tas
+
 class FieldNameMap(Base):
+    """Model for the field_name_map table."""
     __tablename__ = "field_name_map"
 
     field_name_map_id = Column(Integer, primary_key=True)
     table_name = Column(Text)
-    column_to_field_map = Column(Text)
\ No newline at end of file
+    column_to_field_map = Column(Text)
+
+class Appropriation(Base):
+    """Model for the appropriation table."""
+    __tablename__ = "appropriation"
+
+    appropriation_id = Column(Integer, primary_key=True)
+    adjustmentstounobligatedbalancebroughtforward_cpe = Column(Numeric, nullable=False)
+    agencyidentifier = Column(Text, nullable=False)
+    allocationtransferagencyidentifier = Column(Text)
+    availabilitytypecode = Column(Text)
+    beginningperiodofavailability = Column(Text)
+    borrowingauthorityamounttotal_cpe = Column(Numeric)
+    budgetauthorityappropriatedamount_cpe = Column(Numeric, nullable=False)
+    budgetauthorityavailableamounttotal_cpe = Column(Numeric, nullable=False)
+    budgetauthorityunobligatedbalancebroughtforward_fyb = Column(Numeric)
+    contractauthorityamounttotal_cpe = Column(Numeric)
+    deobligationsrecoveriesrefundsbytas_cpe = Column(Numeric, nullable=False)
+    endingperiodofavailability = Column(Text)
+    grossoutlayamountbytas_cpe = Column(Numeric, nullable=False)
+    mainaccountcode = Column(Text, nullable=False)
+    obligationsincurredtotalbytas_cpe = Column(Numeric, nullable=False)
+    otherbudgetaryresourcesamount_cpe = Column(Numeric)
+    spendingauthorityfromoffsettingcollectionsamounttotal_cpe = Column(Numeric)
+    statusofbudgetaryresourcestotal_cpe = Column(Numeric, nullable=False)
+    subaccountcode = Column(Text, nullable=False)
+    unobligatedbalance_cpe = Column(Numeric, nullable=False)
+    tas = Column(Text, index=True, nullable=False, default=concatTas, onupdate=concatTas)
+
+class ObjectClassProgramActivity(Base):
+    """Model for the object_class_program_activity table."""
+    __tablename__ = "object_class_program_activity"
+
+    object_class_program_activity_id = Column(Integer, primary_key=True)
+    agencyidentifier = Column(Text, nullable=False)
+    allocationtransferagencyidentifier = Column(Text)
+    availabilitytypecode = Column(Text)
+    beginningperiodofavailability = Column(Text)
+    bydirectreimbursablefundingsource = Column(Text, nullable=False)
+    deobligationsrecoveriesrefundsprioryrbyprogobjectclass_cpe = Column(
+        Numeric, nullable=False, key="deobligationsrecoveriesrefundsdofprioryearbyprogramobjectclass_cpe")
```

Comment:
And just because that screenshot makes me nervous, they are the same attribute yes?  So if you follow those lines by getting the long name you'll get 333?